robot_skills/src/robot_skills/arms.py

```diff
@@ -455,6 +507,39 @@ def wait_for_motion_done(self, timeout=10.0, cancel=False):
             # return self._ac_gripper.wait_for_result(rospy.Duration(timeout - passed_time))
             return True
 
+    @property
+    def object_in_gripper_state(self):
+        """ Returns whether the gripper is empty, holding an object or if this is unknown
+
+        :return: ObjectInGripperState
+        """
+        distance = self.grasp_sensor_distance
+        if distance is None:
+            return ObjectInGripperState.UNKNOWN
+        elif distance < GRASP_SENSOR_THRESHOLD:
+            return ObjectInGripperState.HOLDING
+        else:
+            return ObjectInGripperState.EMPTY
+
+    @property
+    def grasp_sensor_distance(self):
+        """ Returns the sensor distance. If no recent measurement is available, None is returned
+
+        :return: float with distance, None if no recent measurement
+        """
+        distance, stamp = self._grasp_sensor_state
+        if (rospy.Time.now() - stamp).to_sec() < 0.5:
+            return distance
+        else:
+            return None
+
+    def _grasp_sensor_callback(self, msg):
+        """ Callback function for grasp sensor messages
+
+        :param msg: std_msgs.msg.Float32
+        """
+        self._grasp_sensor_state = GripperMeasurement(msg.data[0])
```

Comment:
What is in the rest of the array ```msg.data```? Wouldn't a plain Float32 make more sense? The messages are a std_msgs.msg.Float32MultiArray according to the constructor

Reply:
You're right twice. I updated the docstring. I couldn't update the msg type though, since this depends on the available Orocos components. Updating this will take a lot of work and is a completely different issue.